src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -118,7 +118,14 @@ sealed trait Task extends Instance {
 }
 
 object Task {
-  /**
+
+  def from(i: Instance): Option[Task] = from(Some(i))
```

Comment:
apply is syntactic sugar for Task(...). I don't know if I see extra value
in "from"
On Thu, Sep 8, 2016 at 8:36 PM James DeFelice <notifications@github.com>
wrote:

> In src/main/scala/mesosphere/marathon/core/task/Task.scala
> <https://github.com/mesosphere/marathon/pull/4336#discussion_r78124703>:
>
> > @@ -118,7 +118,14 @@ sealed trait Task extends Instance {
> >  }
> >
> >  object Task {
> > -  /**
> > +
> > +  def from(i: Instance): Option[Task] = from(Some(i))
>
> `from` seemed more clear to me. If `apply` is more idiomatic that's fine
> … <#m_-2102650954803935586_>
> On Thu, Sep 8, 2016 at 8:35 PM, Jason Gilanfarr ***@***.***> wrote: In
> src/main/scala/mesosphere/marathon/core/task/Task.scala <#4336 (comment)
> <https://github.com/mesosphere/marathon/pull/4336#discussion_r78113348>>:
> > @@ -118,7 +118,14 @@ sealed trait Task extends Instance { > } > > object
> Task { > - /** > + > + def from(i: Instance): Option[Task] = from(Some(i))
> Why not apply? — You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <
> https://github.com/mesosphere/marathon/pull/4336/files/1dbfa88c79764d63f7340e618c5af92001554705#r78113348>,
> or mute the thread <
> https://github.com/notifications/unsubscribe-auth/ACPVLBI-5BgHzGY2j9lukqkg7atoW32xks5qoKnbgaJpZM4J4k2Q>
> .
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4336/files/1dbfa88c79764d63f7340e618c5af92001554705#r78124703>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAweZuTRRKqrqVnyo9Wqm_x3awnbnGbcks5qoNQpgaJpZM4J4k2Q>
> .
>